lib/tasks/akismet.rake

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+desc 'Checks old posts using Akismet'
+task 'akismet:scan_old' => :environment do
+  # Scan the first post of users that match:
+  # - regular posts only
+  # - not deleted posts
+  # - post contains a URL
+  # - user is not trusted (TL0-2 and not staff)
+  # - no other post of this user was checked before
+  sql = <<~SQL
+    WITH first_user_posts AS (
+      SELECT MIN(posts.id) post_id, posts.user_id
+      FROM posts
+      JOIN topics on topics.id = posts.topic_id
+      WHERE (raw LIKE '%http%' OR cooked LIKE '%href%')
+        AND archetype = 'regular'
+        AND posts.deleted_at IS NULL
+        AND topics.deleted_at IS NULL
+      GROUP BY posts.user_id
+    )
+    SELECT user_id, post_id
+    FROM first_user_posts
+    JOIN users u on u.id = first_user_posts.user_id
+    WHERE NOT EXISTS (SELECT 1
+                      FROM post_custom_fields c
+                      JOIN posts ON posts.id = c.post_id
+                      WHERE posts.user_id = first_user_posts.user_id
+                        AND name = 'AKISMET_STATE'
+                      ) -- no post from this user was checked or is scheduled
```

Comment:
I am not so sure about this condition. I've seen spam users having a legit comment and then starting to spam with URLs. If we omitted this condition, then we'd check the first post containing a URL, which I find to be a better heuristic.

Reply:
Sure, not checking the same post more than once is a good idea 👍